src/compass/utils/runconfig.py

```diff
@@ -0,0 +1,280 @@
+from sentinel1_reader.sentinel1_orbit_reader import get_swath_orbit_file_from_list
+from sentinel1_reader.sentinel1_reader import burst_from_zip
+
+
+def validate_group(group_cfg: dict) -> None:
+    """Check and validate runconfig entries.
+
+    Parameters
+    ----------
+    group_cfg : dict
+        Dictionary storing runconfig options to validate
+    """
+    error_channel = journal.error('runconfig.validate_group')
+
+    # Check 'input_file_group' section of runconfig
+    input_group = group_cfg['input_file_group']
+    # If is_reference flag is False, check that file path to reference
+    # burst is assigned and valid (required by geo2rdr and resample)
+    is_reference = input_group['reference_burst']['is_reference']
+    if not is_reference:
+        helpers.check_file_path(input_group['reference_burst']['file_path'])
+
+    safe_pol_modes = []
+    for safe_file in input_group['safe_file_path']:
+        # Check if files exists
+        helpers.check_file_path(safe_file)
+
+        # Get safe pol mode
+        safe_pol_modes.append(helpers.get_file_mode(safe_file))
+
+    for orbit_file in input_group['orbit_file_path']:
+        helpers.check_file_path(orbit_file)
+
+    # Check 'dynamic_ancillary_file_groups' section of runconfig
+    # Check that DEM file exists and is GDAL-compatible
+    dem_path = group_cfg['dynamic_ancillary_file_group']['dem_file']
+    helpers.check_file_path(dem_path)
+    helpers.check_dem(dem_path)
+
+    # Check 'product_path_group' section of runconfig.
+    # Check that directories herein have writing permissions
+    product_path_group = group_cfg['product_path_group']
+    helpers.check_write_dir(product_path_group['product_path'])
+    helpers.check_write_dir(product_path_group['scratch_path'])
+    helpers.check_write_dir(product_path_group['sas_output_file'])
+
+    # Check polarizations to be processed.
+    have_default_pol = 'polarization' in group_cfg['processing']
+    if have_default_pol:
+        # Determine allowable SAFE polarization modes
+        cfg_pols = group_cfg['processing']['polarization']
+        cfg_pol_modes = []
+        if ('HH' in cfg_pols and 'HV' in cfg_pols) or ['HH'] == cfg_pols:
+            cfg_pol_modes = ['DH', 'SH']
+        if ('VV' in cfg_pols and 'VH' in cfg_pols) or ['VV'] == cfg_pols:
+            cfg_pol_modes = ['DV', 'SV']
+        # Check YAML has valid polarization
+        if not cfg_pol_modes:
+            err_str = f"Invalid polarization combo: {cfg_pols}"
+            error_channel.log(err_str)
+            raise ValueError(err_str)
+
+        # Check for SAFE files with non-matching polarization
+        mode_mismatch = any([safe_mode not in cfg_pol_modes
+                             for safe_mode in safe_pol_modes])
+        if mode_mismatch:
+            err_str = "Unexpected safe file polarization mode"
+            error_channel.log(err_str)
+            raise ValueError(err_str)
```

Comment:
The logic for polarization should be consistent with what I commented [here](https://github.com/opera-adt/COMPASS/pull/4#discussion_r797139128) and [here](https://github.com/opera-adt/COMPASS/pull/4#discussion_r786960268).